components/core/src/origin.rs

```diff
@@ -0,0 +1,89 @@
+use crate::{error::Error,
+            package::ident::is_valid_origin_name};
+use serde_derive::{Deserialize,
+                   Serialize};
+use std::{fmt,
+          result,
+          str::FromStr};
+
+pub const ORIGIN_MEMBER_ROLE_READONLY_MEMBER: &str = "readonly_member";
+pub const ORIGIN_MEMBER_ROLE_MEMBER: &str = "member";
+pub const ORIGIN_MEMBER_ROLE_MAINTAINER: &str = "maintainer";
+pub const ORIGIN_MEMBER_ROLE_ADMINISTRATOR: &str = "administrator";
+pub const ORIGIN_MEMBER_ROLE_OWNER: &str = "owner";
```

Comment:
No need to change anything, but I am personally a fan of asciating constants like this with the type. So something like:

```rust
impl OriginMemberRole {
    const READONLY_MEMBER: &'static str = "readonly_member";
    const MEMBER: &'static str = "member";
    const MAINTAINER: &'static str = "maintainer";
    const ADMINISTRATOR: &'static str = "administrator";
    const OWNER: &'static str = "owner";
}
```

You can then access them with `OriginMemberRole::READONLY_MEMBER`

Reply:
Totally agree. Adjusted in https://github.com/habitat-sh/habitat/pull/7841/commits/9aee65a4697d2a584ada86c503987c6b5abd8120

Reply:
I like your change. Although, it was not quite what I had in mind :smile:. My intention was to scope the constants within the `OriginMemberRole` so they should be defined in an `impl OriginMemberRole { }` block and accessed with `OriginMemberRole::READONLY_MEMBER`. But I dont think there is any need to change anything for this case.

Reply:
ahh yes. I'll make it so.

Reply:
I've modified in https://github.com/habitat-sh/habitat/pull/7841/commits/78f9d5e6a2b76e641bb6f2a2971c62744b5e762b